packages/browser/src/plugins/remote-loader/index.ts

```diff
@@ -17,6 +25,85 @@ export interface RemotePlugin {
   settings: JSONObject
 }
 
+export class ActionDestination implements Plugin {
+  name: string // destination name
+  version = '1.0.0'
+  type: Plugin['type']
+
+  alternativeNames: string[] = []
+
+  middleware: DestinationMiddlewareFunction[] = []
+
+  action: Plugin
+
+  constructor(name: string, action: Plugin) {
+    this.action = action
+    this.name = name
+    this.type = action.type
+    this.alternativeNames.push(action.name)
+  }
+
+  addMiddleware(...fn: DestinationMiddlewareFunction[]): void {
+    this.middleware.push(...fn)
+  }
+
+  private async transform(ctx: Context): Promise<Context> {
+    const modifiedEvent = await applyDestinationMiddleware(
+      this.name,
+      klona(ctx.event),
```

Comment:
Why is this serialize/deserialize neccessary? sometimes its a hacky way of deserializing a class, but the event is just an object, right?

Reply:
We do need to mutate the event I believe, but I moved the cloning logic into applyDestinationMiddleware and added a test that will fail if the original context gets modified (ie. if klona gets removed) 👍